maven-model-builder/src/test/java/org/apache/maven/model/building/CyclicParentTest.java

```diff
@@ -0,0 +1,143 @@
+package org.apache.maven.model.building;
+
+  /*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.io.File;
+import java.util.Properties;
+
+import junit.framework.TestCase;
+import org.apache.maven.building.StringSource;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.Parent;
+import org.apache.maven.model.Repository;
+import org.apache.maven.model.resolution.InvalidRepositoryException;
+import org.apache.maven.model.resolution.ModelResolver;
+import org.apache.maven.model.resolution.UnresolvableModelException;
+import org.apache.maven.model.resolution.WorkspaceModelResolver;
+
+/**
+ * @author Guillaume Nodet
+ */
+public class CyclicParentTest
```

Comment:
It seems that JUnit 4.13.1 is used in the project. Could this TC be upgraded to JUnit 4 style?
Perhaps it could then use ` @Test(expected=ModelBuildingException.class)` instead of comment in `testCyclic`.

Reply:
It looks like test for `CyclicParent`, but there is nothing named so in this project. Could this be renamed to make it clear which unit (`ModelBuilder` I guess) is under the test?

Reply:
@pzygielo unfortunately, that does not seem to be sufficient and a quick scan shows me that this pattern is not used anywhere in the maven source tree.  Maybe this require a surefire upgrade for it to work ?
Anyway, given it does not seem to work, I suppose it requires a different issue to fix all similar patterns, for example see https://github.com/apache/maven/blob/618e62dd3315b4cb5b0f7dcdd37fc787c44b2ade/maven-builder-support/src/test/java/org/apache/maven/building/FileSourceTest.java#L34-L46

Reply:
I renamed and improved the test class a bit.

Reply:
> @pzygielo unfortunately, that does not seem to be sufficient and a quick scan shows me that this pattern is not used anywhere in the maven source tree. Maybe this require a surefire upgrade for it to work ?
> Anyway, given it does not seem to work, I suppose it requires a different issue to fix all similar patterns, for example see
> 
> https://github.com/apache/maven/blob/618e62dd3315b4cb5b0f7dcdd37fc787c44b2ade/maven-builder-support/src/test/java/org/apache/maven/building/FileSourceTest.java#L34-L46

Ah, the problem was that the test class was extending `TestCase` which was causing the test to be run with a Unit 3 runner...  This should be fixed now.

Reply:
Ah, and I switched to the recommended `assertThrows()` instead so that it is compatible with JUnit 5.